### ISSUE TITLE
Unable to use cuda_NvidiaOpticalFlow_1_0

### ISSUE THREAD

**Issue by ganessh22** [opencv/opencv#26168]
### System Information

OpenCV 4.7.0
Compiled with CUDA capabilities. 
On Docker using nvidia CUDA 12.3 cudnn image

### Detailed description

I am trying to switch from `cuda.FarnebackOpticalFlow` to `cuda.cuda_NvidiaOpticalFlow_1_0`
I have two images each of (720, 1280, 3) shape on GPU as `cv2.cuda_GpuMat`s. 

```python
height, width = current_frame.size()
nvof = cv2.cuda_NvidiaOpticalFlow_1_0.create((width, height), 5, False, False, False, 0)
> flow_down = nvof.calc(current_frame, offset_frame, None)
flow = nvof.upSampler(flow_down[0], width, height, nvof.getGridSize(), None)        
```

this line fails with the error
```bash
flow_down = nvof.calc(current_frame, offset_frame, None)
cv2.error: OpenCV(4.7.0) /opencv-4.7.0/modules/core/src/cuda/gpu_mat.cu:281: error: (-217:Gpu API call) invalid pitch argument in function 'copyTo'
```

if I modify the code to use ndarray as input
```python
height, width = current_frame.size()
nvof = cv2.cuda_NvidiaOpticalFlow_1_0.create((width, height), 5, False, False, False, 0)
> flow_down = nvof.calc(current_frame.download(), offset_frame.download(), None)
flow = nvof.upSampler(flow_down[0], width, height, nvof.getGridSize(), None)        

```
I get a similar error
```bash
*** cv2.error: OpenCV(4.7.0) /opencv-4.7.0/modules/core/src/cuda/gpu_mat.cu:233: error: (-217:Gpu API call) invalid pitch argument in function 'upload'
```
I can use other OpticalFlow methods fine such as below with GpuMat as input
```python
farneback = cv2.cuda.FarnebackOpticalFlow_create()
farneback.calc(
    current_frame,
    offset_frame,
    None,
)
```
Is there a working Python sample for this? Or please help solve this issue?

### Steps to reproduce

I am trying to switch from `cuda.FarnebackOpticalFlow` to `cuda.cuda_NvidiaOpticalFlow_1_0`
I have two images each of (720, 1280, 3) shape on GPU as `cv2.cuda_GpuMat`s. 

```python
height, width = current_frame.size()
nvof = cv2.cuda_NvidiaOpticalFlow_1_0.create((width, height), 5, False, False, False, 0)
> flow_down = nvof.calc(current_frame, offset_frame, None)
flow = nvof.upSampler(flow_down[0], width, height, nvof.getGridSize(), None)        
```

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [ ] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by cudawarped** [comment]
I suspect your issue is here
> height, width = current_frame.size()

because size is returned as **(width, height)**.

> Is there a working Python sample for this? Or please help solve this issue?

There is a python sample test folder of the module

https://github.com/opencv/opencv_contrib/blob/4.x/modules/cudaoptflow/misc/python/test/test_nvidiaopticalflow.py

**Comment by ganessh22** [comment]
Thank you. It works now. It was the height width interchange and some other issues which I fixed with the file you linked. 
Just a related question.
Still is slower than Farneback which is strange. https://developer.nvidia.com/blog/opencv-optical-flow-algorithms-with-nvidia-turing-gpus/ shows much faster. Even if I change the argument NV_PERF_LEVEL to fast from 5 to 10 or even 20. Not sure what GPU they used to get the numbers in the post but I see 22 - 13 ms for NVOF vs 12 ms for Farneback.

**Comment by cudawarped** [comment]
>Still is slower than Farneback

How are you timing it?  Are you timing over a number of frames ignoring the initialization cost when processing the first few frames (e.g. only timing execution of processing frames 5-30).  If so try using CUDA events, again ignoring the initialization cost or use [Nsight Compute](https://developer.nvidia.com/nsight-compute).

**Comment by ganessh22** [comment]
I was ignoring the first two frames as I saw the first run was always much slower. Let me try again with CUDA events. I've seen sometimes using time.time for GPU stuff gives wrong benchmarks.
